packages/shell-api/src/database.ts

```diff
@@ -1063,7 +1063,10 @@ export default class Database extends ShellApiWithMongoClass {
     }
 
     const adminDb = this.getSiblingDB('admin');
-    const aggregateOptions = { $readPreference: { mode: 'primaryPreferred' } };
+    const aggregateOptions = {
+      $readPreference: { mode: 'primaryPreferred' },
+      bsonRegExp: true,
```

Comment:
That was it? Does this deserve a comment in code?

Reply:
Yes, from my understanding it was just a matter of passing the correct set options in this case. I'll add a comment.